_rules/table-headers-attribute-refer-to-data-cells.md

```diff
@@ -0,0 +1,223 @@
+---
+id: a25f45
+name: `headers` attribute only refers to `cells` in the same `table` element
+test_type: atomic
+description: |
+  This rule checks that the `headers` attribute must only refer to `cells` in the same `table` element
+test_aspects:
+- DOM Tree
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Test Procedure
+
+### Applicability
+
+This rule applies to `td` and `th` elements with a `headers` attribute and within a `table` element with a [semantic role](#semantic-role) of `table` or `grid` or `treegrid`, that is [included in the accessibility tree](#included-in-the-accessibility-tree).
```

Comment:
If we scope `grid` and `treegrid` don't we need to scope to `columnheader` and `rowheader` too ? (https://www.w3.org/TR/wai-aria-practices-1.1/#treegrid_roles_states_props  - https://www.w3.org/TR/wai-aria-practices-1.1/#grid_roles_states_props). 

Can we just focus on role `table` ?

Do we need to specify too that th needs to not have an explicit role define ? td and th with no explicit semantic role or the same as the implicit role  ?

> This rule applies to `td` and `th` elements with a `headers` attribute and within a `table` element with a [semantic role](#semantic-role) of `table` that is [included in the accessibility tree](#included-in-the-accessibility-tree). `td` and `th` elements has no explicit semantic role or an explicit semantic role that is identical to their explicit semantic role.

Reply:
I agree with @audreymaniez suggestions. Just some corrections to the last sentence of the proposed applicability:
`td` and `th` elements have no explicit semantic role or an explicit semantic role that is identical to their implicit semantic role.

Reply:
We discussed a little internally in our office, and would suggest that we in this rule look at the semantic roles of elements (table, grid, rowheader, columnheader, cell), instead of the html element names (table, th, td).
This will be much in line with most other rules we have written.

And then, like we have discussed previously, we would suggest that we don't limit the scope of the rule to cells that have a `headers` attribute, but in general just check whether a header can be calculated for that particular cell (there is an algorithm for it in the html spec: https://www.w3.org/TR/html/tabular-data.html#forming-relationships-between-data-cells-and-header-cells).

@kasperisager also had the point that if we want to catch the roles `table`, `grid` and `treegrid`, you can just say something like "with the semantic role of `table` or a semantic role that inherits from the `table` role".
Then if you really want to list the roles too, you can have a note saying: "The roles that inherit from the `table` role are `grid` and `treegrid`". 
But in this way, it will be clear what those particular roles have in common and why they are in this rule together.

Reply:
Based on the discussion with @WilcoFiers & @annethyme, we concluded that we will scope this to only `table` elements. I have made some updates to the rule.